src/helpers/client.js

```diff
@@ -1,10 +1,49 @@
-import axios from 'axios';
+import axios, { axiosCall } from 'axios';
+
+import { loginPageLink } from '../constants/link.constants';
+import { history } from '../helpers/history';
 
 require('dotenv').config();
 
-export default axios.create({
+const client = axios.create({
   baseURL: process.env.REACT_APP_BASE_URL,
   headers: {
     'Content-type': 'application/json',
   },
 });
+
+client.interceptors.response.use(
+  (res) => res,
+  async (error) => {
+    if (
+      error.config.url != 'rest-auth/login/' &&
+      error.config.url != 'rest-auth/registration/' &&
+      error.config.url != 'rest-auth/logout/'
+    ) {
+      if (error.response.status == 401) {
+        const refreshToken = localStorage.getItem('refreshToken');
+        if (refreshToken) {
```

Comment:
Just a question, are you refreshing the token even if you have a non-expired one? I think you should have an endpoint to validate the token first, don't you?

Reply:
I refresh only when the response status is 401, that is, unauthorized. So if I am authorized there is no problem about that